### ISSUE TITLE
Suppression for RV bug pattern not evaluated (for a lambda)

### ISSUE THREAD

**Issue by uhafner** [spotbugs/spotbugs#1007]
I have several code examples in the style of issue #872:  

```
    @Test @SuppressFBWarnings("RV")
    void shouldThrowExceptionOnWrongIndex() {
        Report report = new Report();
        report.addAll(asList(HIGH, NORMAL_1, NORMAL_2));

        assertThatThrownBy(() -> report.get(-1))
                .isInstanceOf(IndexOutOfBoundsException.class)
                .hasMessageContaining("-1");
        assertThatThrownBy(() -> report.get(3))
                .isInstanceOf(IndexOutOfBoundsException.class)
                .hasMessageContaining("3");
    }
```

Spotbugs complains that a bug type `RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT` has been found. In order to suppress this warning I annotated the method with `@SuppressFBWarnings("RV")` (or `@SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT)`), however the warning still is present, see [CI Result](https://ci.jenkins.io/job/Plugins/job/analysis-model/job/master/708/spotbugs/source.73150961-98f6-4d7a-a209-91555bd7d902/#563). Seems that the annotation is not evaluated. I have other code examples where this annotation correctly suppresses a bug.

**Comment by jonathan-lmax** [comment]
I just ran across this, it happens when the bug to suppress is inside a lambda as this is technically a different method.

**Comment by jerith666** [comment]
#724 is the issue for suppressing warnings inside lambdas.
